### ISSUE TITLE
AgdaStdlib fails to build on darwin

### ISSUE THREAD

**Issue by mcwitt** [NixOS/nixpkgs#87608]
**Describe the bug**
Building `nixpkgs.AgdaStdlib` fails on darwin with latext nixpkgs-unstable:
``` sh
$ git rev-parse HEAD
cfe68f2b68b7a7c2e5347c496b6963af30d18d3f
```

**To Reproduce**
``` sh
$ nix-build -A pkgs.AgdaStdlib
```
fails with
```
B.b != q₁ of type Level
when checking that the inferred type of an application
  Any.map _f_1489 y∈ys ≡ pq′
matches the expected type
  Any.map (λ q₂ → p , q₂) (Any.map (λ x₁ → P.subst Q x₁ _1452) y∈ys)
  ≡ pq′
builder for '/nix/store/vfslyvapzgsmja2w04ssyzqgbbdw9v7g-agda-stdlib-1.1.drv' failed with exit code 1
error: build of '/nix/store/vfslyvapzgsmja2w04ssyzqgbbdw9v7g-agda-stdlib-1.1.drv' failed
```

**Expected behavior**
A successful build.

**Additional context**
This might have been fixed upstream: https://github.com/agda/agda/issues/4351

**Notify maintainers**
@jwiegley 
@laMudri 

**Metadata**
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.4.0, macOS 10.15.4`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.4`
 - channels(matt): `"darwin, home-manager"`
 - channels(root): `"nixpkgs-20.09pre221814.10100a97c89, darwin, home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by laMudri** [comment]
Probably the mismatch between old stdlib (current version is 1.3) and new Agda. Most probably will be fixed by #76653, which is looking close to a merge. The workaround for now is to clone your own version of stdlib at tag v1.3, and use Agda's ordinary library management directly.

**Comment by mcwitt** [comment]
Thanks, the new infrastructure in #76653 is looking promising! Using `agda.withPackages` on that branch was a solution for me: https://github.com/mcwitt/plfa.github.io/blob/8ea3b50ddc2ab2355f19152bfbd683747296d1a2/shell.nix
